core/src/main/java/org/mule/TransformationService.java

```diff
@@ -84,7 +86,7 @@ public MuleMessage applyTransformers(final MuleMessage message, final MuleEvent
      * @param outputType the required output type for this transformation. By adding this parameter some additional
      * transformations will occur on the message payload to ensure that the final payload is of the specified type.
      * If no transformers can be found in the registry that can transform from the return type of the transformation
-     * list to the outputType and exception will be thrown
+     * list to the outputType and exception will be thrown.  This parameter is optional.
```

Comment:
This method is a bit weird, because it receives a list of transformers and then an optional datatype. When a datatype is given, that means that another transformation will be required. Is not more readable to have two different methods, one receiving a lit of transformers an another one receiving a datatype?